### DISCUSSION TITLE
How to run every single query on seperate isolate without MoorIsolate

### DISCUSSION THREAD

**Discussion by salman731** [simolus3/drift#1345] [Q&A]
I want to ask that how to run query on separate isolate and use it again without creating new isolate without isolate.

**Comment by simolus3** [comment]
You can open a moor database on another isolate that you control - no problem. `MoorIsolate` is a convenience layer which hides away that queries are running on a background isolate. Even though can't share complex objects across an isolate, `MoorIsolate` makes it seem as if the same database instance was active on multiple isolates.

I don't really understand your question. You want to run a query on a separate isolate without using a `MoorIsolate`? In that case you'd have to manually open another database instance on a background isolate that you control.

**Reply by salman731** [reply]
I want to run isolate on insert data...

**Reply by davidmartos96** [reply]
If you are experiencing performance issues with many inserts at the same time, it's very likely that wrapping them inside a transaction will improve it. Maybe you don't need isolates at all

**Reply by salman731** [reply]
can I use Isolate and transaction together?

**Reply by simolus3** [reply]
You can open the database on another isolate, start a transaction there and insert all the data inside a single transaction. But I'm wondering why you wouldn't want to use a `MoorIsolate` if you end up using a background isolate either way?
